fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/IndexState.java

```diff
@@ -55,7 +55,15 @@ public enum IndexState {
      * reads or queries and it is not maintained
      * by the record store.
      */
-    DISABLED(2L, "indexesDisabled");
+    DISABLED(2L, "indexesDisabled"),
+    /**
+     * Indicates that this unique index is fully "indexed", but
+     * some uniqueness violations still exist. This may happen
+     * when the online indexer finds some duplicating records.
+     * In this mode, it is safe to consider an index as {@link #READABLE}
+     * for queries as long as uniqueness is not assumed.
+     */
+    READABLE_UNIQUE_PENDING(3L, "indexesReadableUniquePending");
```

Comment:
I am wondering whether there is a process for transitioning the default, to allow this state. I don't think we currently have a process for this, so coming up with one as part of this change doesn't really make sense.

Reply:
Since this state will appear in the DB after the user will actively allow it during indexing, would your second suggestion (depending on users not setting it until after all instances are updated) be acceptable? 
If so, I can add a warning to the proper javadoc (`shouldAllowUniquePendingState`).